tests/net/checksum_offload/src/main.c

```diff
@@ -294,15 +294,15 @@ ETH_NET_DEVICE_INIT(eth1_offloading_disabled_test,
 		    "eth1_offloading_disabled_test",
 		    eth_init, NULL,
 		    &eth_context_offloading_disabled, NULL,
-		    CONFIG_ETH_INIT_PRIORITY,
+		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
```

Comment:
If a test is creating Ethernet device (like here), then we should use the `CONFIG_ETH_INIT_PRIORITY`. If we have disabled Ethernet in the config file, but are still using Ethernet configs, then we should fix the test to not use Ethernet init macros and configs. Just blindly changing to use different config option is not a proper fix.

Reply:
If I understand this correctly `CONFIG_ETH_INIT_PRIORITY` is a priority for Ethernet drivers since it is defined there (and probably should be named as CONFIG_ETH_DRIVER_INIT_PRIORITY`.
We disable Ethernet drivers and we should not have access to this configuration option.
However we have access to driver API, so it is OK to use driver API in tests.

Reply:
The network tests have their own Ethernet driver, that is not disabled. Only the Ethernet drivers in drivers/ethernet directory are disabled by that option so that they are not accidentally used when the test is ran in a real board.